server/services/v1/admin.go

```diff
@@ -0,0 +1,78 @@
+// Copyright 2022 Tigris Data, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package v1
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+
+	"github.com/fullstorydev/grpchan/inprocgrpc"
+	"github.com/go-chi/chi/v5"
+	"github.com/grpc-ecosystem/grpc-gateway/v2/runtime"
+	api "github.com/tigrisdata/tigris/api/server/v1"
+	"github.com/tigrisdata/tigris/server/metadata"
+	"github.com/tigrisdata/tigris/server/transaction"
+	"google.golang.org/grpc"
+)
+
+const (
+	adminPath = "/admin/" + version
+)
+const (
+	namespacePath        = "/namespaces"
+	namespacePathPattern = namespacePath + "/*"
+)
+
+type adminService struct {
+	api.UnimplementedAdminServer
+	tenantMgr *metadata.TenantManager
+	txMgr     *transaction.Manager
+}
+
+func newAdminService(tenantMgr *metadata.TenantManager, txMgr *transaction.Manager) *adminService {
+	return &adminService{
+		tenantMgr: tenantMgr,
+		txMgr:     txMgr,
+	}
+}
+
+func (a *adminService) CreateNamespace(ctx context.Context, req *api.CreateNamespaceRequest) (*api.CreateNamespaceResponse, error) {
+	namespace := metadata.NewTenantNamespace(req.Name, uint32(req.Id))
+	_, err := a.tenantMgr.CreateTenant(ctx, a.txMgr, namespace)
+	if err != nil {
+		return nil, err
+	}
+	return &api.CreateNamespaceResponse{
```

Comment:
This type doesn't resolve for me, either, FYI. Not sure if we want to update the submodule references twice, once during development and once when API was merged and we are ready to commit this back to main?

Reply:
did you try make generate? if it still persists, try invalidating IDE cache.